app/models/miq_report.rb

```diff
@@ -97,6 +97,10 @@ def col_format_with_defaults
     end
   end
 
+  def conditions=(exp)
+    super(exp.nil? || exp.kind_of?(MiqExpression) ? exp : MiqExpression.new(exp))
```

Comment:
I like your way better, but I think you can also do the following...not sure which is more Rubyish

```ruby
    exp = MiqExpression.new(exp) if exp && !exp.kind_of?(MiqExpression)
    super
```

Reply:
I can do that. I copy/pasted the existing code